examples/nlp/semantic_similarity_with_keras_nlp.py

```diff
@@ -0,0 +1,274 @@
+    epochs: The number of epochs to train the model
+    batch_size: The number of samples per gradient update
+"""
+
+bert_clf.fit(
+    x=x_train, y=y_train, validation_data=(x_val, y_val), epochs=1, batch_size=512
+)
+
+"""
+
+Our BERT classifier gave us ~68% accuracy on validation split, let's see how it performs
+on Test split
+
+### Evaluate the performance of the trained model on test data.
+"""
+bert_clf.evaluate(x=x_test, y=y_test)
+
+"""
+Our baseline bert gave almost similar (~68%) accuracy on the test split. Let's see if we can
+improve it.
+
+Let's recompile our model with a different learning rate and see performance
+"""
+bert_clf_1 = keras_nlp.models.BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3, preprocessor=None
+)
+
+bert_clf_1.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=tf.keras.optimizers.Adam(1e-4),
+    metrics=["accuracy"],
+)
+
+bert_clf_1.fit(
+    x=x_train, y=y_train, validation_data=(x_val, y_val), epochs=1, batch_size=512
+)
+
+bert_clf_1.evaluate(x=x_test, y=y_test)
+"""
+This time we got 72% accuracy on val and test split. Not bad for 1 epoch ! Let's save our model
+for now and learn how to perform inference with it. We took batch size of 512 to utilize our GPUs fully.
+
+# Save and Reload the model
+"""
+bert_clf.save("bert_classifier.pb")
+restored_model = keras.models.load_model("bert_classifier.pb")
+restored_model.evaluate(x=x_test, y=y_test)
+
+"""
+# Inference
+
+Randomly sample 4 sentence pairs from the test set
+"""
+infer_df = to_dataframe(snli_test).sample(n=4)
+infer_df.head()
+
+# Making predictions
+preprocessed = bert_preprocessor((infer_df["hypothesis"], infer_df["premise"]))
+predictions = bert_clf.predict(preprocessed)
+print(tf.math.argmax(predictions, axis=1).numpy())
+
+"""
+# Enhancing accuracy with RoBERTA
+
+Now that we have established a baseline, we'll attempt to get better results by trying different
+models. KerasNLP makes experimentation easy for us, with just few lines of code, we can train 
+a roberta on the same dataset.
+"""
+
+# Roberta has it's own data preprocessing methods
+roberta_preprocessor = keras_nlp.models.RobertaPreprocessor.from_preset(
+    "roberta_base_en"
+)
+
+# we preprocess original snli dataset with roberta preprocessor
+x_train, y_train = preprocess_dataset(snli_train, roberta_preprocessor)
+x_test, y_test = preprocess_dataset(snli_test, roberta_preprocessor)
+x_val, y_val = preprocess_dataset(snli_val, roberta_preprocessor)
+
+# inittializing a roberta from preset
+roberta_clf = keras_nlp.models.RobertaClassifier.from_preset(
+    "roberta_base_en", num_classes=3, preprocessor=None
+)
+
+roberta_clf.summary()
+
+roberta_clf.fit(
+    x=x_train, y=y_train, validation_data=(x_val, y_val), epochs=1, batch_size=16
+)
+
+roberta_clf.evaluate(x=x_test, y=y_test)
+
+"""
+`robeta_base_en` is slightly bigger model than bert_tiny, it took almost 1.5 hrs to train on
```

Comment:
roberta_base_en is a much bigger model right? We might want to call out exactly how much the parameter counts are changing, I think you are growing the model 30-fold!